app/experimenter/nimbus-ui/README.md

```diff
@@ -75,6 +75,106 @@ export interface GetExperimentOverviews {
 
 **Note**: this command performs introspection on the GraphQL endpoint, so your local server must be running for it to work.
 
+## Error handling
+
+This app has an [`AppErrorBoundary`](./src/components/AppErrorBoundary/index.tsx) that will capture any uncaught error that occurs and report them to Sentry. This acts as our last line of defense, but ideally we are able to handle errors before they get to this stage. As well, errors that we wish to display to the user should generally be handled in a consistent fashion.
+
+There are two types of errors to account for:
+
+### Validation errors
+
+These occur when you try to create or modify a record and the server tells us that one or more fields provided incorrect data. For these you should use React Bootstrap's [`Form.Control.Feedback`](https://react-bootstrap.github.io/components/forms/#form-control-feedback-props) component in combination with other form components.
+
+A basic example might look like this:
+
+```tsx
+<Form.Group controlId="description">
+  <Form.Label>Description</Form.Label>
+  <Form.Control type="text" autoFocus />
+  <Form.Text>This is the description.</Form.Text>
+  {errors["description"] && (
+    <Form.Control.Feedback type="invalid">
+      {errors["description"].message}
+    </Form.Control.Feedback>
+  )}
+</Form.Group>
+```
+
+### General errors
+
+General errors can occur when an operation fails altogether. This could be for any reason, such as when a network request fails, or a bad code path raises an exception. For these errors you'll use React Bootstrap's [`Alert`](https://react-bootstrap.github.io/components/alerts/) components.
+
+For example, if error occurs while saving a record:
+
+```tsx
+<Alert variant="warning">Sorry, there was a problem.</Alert>
+```
+
+_Note:_ displaying technical error messages to the user may not be very helpful, so it's generally advised to either provide a [generic message](./src/lib/constants.ts) or instructions on what to do.
```

Comment:
We do return `<Alert>` in `AppErrorBoundary`; is that the only place we might need this? I'm not sure when else a "general" error might come up.

Reply:
A general error could occur during a network request, such as `NetworkError`. Noted a little further up:

> This could be for any reason, such as when a network request fails, or a bad code path raises an exception.

Reply:
I suppose I'm wondering where else/when we'd use `<Alert>` in the code. Doesn't it bubble up to `AppErrorBoundary`?